File: examples/go/apigateway_put_method.go
// Añade un metodo GET a un recurso, con respuesta mock (sin Lambda todavia).
// Uso: go run apigateway_put_method.go <api-id> <resource-id>
package main

import (
	"context"
	"fmt"
	"log"
	"os"

	"github.com/aws/aws-sdk-go-v2/aws"
	"github.com/aws/aws-sdk-go-v2/config"
	"github.com/aws/aws-sdk-go-v2/credentials"
	"github.com/aws/aws-sdk-go-v2/service/apigateway"
	"github.com/aws/aws-sdk-go-v2/service/apigateway/types"
)

func main() {
	if len(os.Args) < 3 {
		log.Fatal("Uso: go run apigateway_put_method.go <api-id> <resource-id>")
	}
	apiID := os.Args[1]
	resourceID := os.Args[2]

	ctx := context.Background()
	cfg, err := config.LoadDefaultConfig(ctx,
		config.WithRegion("us-east-1"),
		config.WithCredentialsProvider(credentials.NewStaticCredentialsProvider("test", "test", "")),
	)
	if err != nil {
		log.Fatalf("Error cargando configuracion: %v", err)
	}
	client := apigateway.NewFromConfig(cfg, func(o *apigateway.Options) {
		o.BaseEndpoint = aws.String("http://localhost:4566")
	})

	_, err = client.PutMethod(ctx, &apigateway.PutMethodInput{
		RestApiId:          aws.String(apiID),
		ResourceId:         aws.String(resourceID),
		HttpMethod:         aws.String("GET"),
		AuthorizationType:  aws.String("NONE"),
	})
	if err != nil {
		log.Fatalf("Error creando el metodo: %v", err)
	}
	fmt.Println("Metodo GET creado en el recurso.")

	_, err = client.PutIntegration(ctx, &apigateway.PutIntegrationInput{
		RestApiId:        aws.String(apiID),
		ResourceId:       aws.String(resourceID),
		HttpMethod:       aws.String("GET"),
		Type:             types.IntegrationTypeMock,
		RequestTemplates: map[string]string{"application/json": `{"statusCode": 200}`},
	})
	if err != nil {
		log.Fatalf("Error configurando la integracion: %v", err)
	}
	fmt.Println("Integracion MOCK configurada. Para conectar una Lambda real, usa integracion AWS_PROXY.")
}

File: examples/rust/apigateway_put_method.rs
// Añade un metodo GET a un recurso, con respuesta mock (sin Lambda todavia).
// Uso: cargo run --bin apigateway_put_method -- <api-id> <resource-id>

use aws_config::BehaviorVersion;
use aws_credential_types::Credentials;
use aws_sdk_apigateway::{config::Region, types::IntegrationType, Client, Config};

#[tokio::main]
async fn main() -> Result<(), aws_sdk_apigateway::Error> {
    let args: Vec<String> = std::env::args().collect();
    if args.len() < 3 {
        eprintln!("Uso: cargo run --bin apigateway_put_method -- <api-id> <resource-id>");
        std::process::exit(1);
    }
    let api_id = &args[1];
    let resource_id = &args[2];

    let config = Config::builder()
        .behavior_version(BehaviorVersion::latest())
        .endpoint_url("http://localhost:4566")
        .region(Region::new("us-east-1"))
        .credentials_provider(Credentials::new("test", "test", None, None, "floci"))
        .build();
    let client = Client::from_conf(config);

    client
        .put_method()
        .rest_api_id(api_id)
        .resource_id(resource_id)
        .http_method("GET")
        .authorization_type("NONE")
        .send()
        .await?;
    println!("Metodo GET creado en el recurso.");

    client
        .put_integration()
        .rest_api_id(api_id)
        .resource_id(resource_id)
        .http_method("GET")
        .r#type(IntegrationType::Mock)
        .request_templates("application/json", r#"{"statusCode": 200}"#)
        .send()
        .await?;
    println!("Integracion MOCK configurada. Para conectar una Lambda real, usa integracion AWS_PROXY.");
    Ok(())
}
